libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/RouteLineResources.kt

```diff
@@ -349,9 +362,13 @@ class RouteLineResources private constructor(
         fun build(): RouteLineResources {
             val theRouteLineColorResources: RouteLineColorResources =
                 routeLineColorResources ?: RouteLineColorResources.Builder().build()
-
+            val theCongestionRange = congestionRange ?: MapboxCongestionRange()
+            if (!theCongestionRange.isRangeValid()) {
```

Comment:
I don't think this builder should have the responsibility of validating the CongestionRange object. I suggest creating a factory or builder that does the validating. Make it impossible to create an invalid CongestionRange object to begin with. This way the entire codebase can trust it without having to worry about if it's valid.